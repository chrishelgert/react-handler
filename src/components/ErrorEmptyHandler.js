import React, { PropTypes } from 'react';
import EmptyHandler from './EmptyHandler';
import ErrorHandler from './ErrorHandler';

/**
 * This Sub-Handler combines the Error and Empty-Handler
 */
const ErrorEmptyHandler = ({
  errorMessage,
  ErrorComponent,
  emptyMessage = '',
  emptyPropertyKey,
  EmptyComponent,
  componentProps = {},
  Component,
}) => (
  <ErrorHandler
    message={errorMessage}
    ErrorComponent={ErrorComponent}
    Component={() => (
      <EmptyHandler
        message={emptyMessage}
        emptyPropertyKey={emptyPropertyKey}
        EmptyComponent={EmptyComponent}
        componentProps={componentProps}
        Component={Component}
      />
    )}
  />
);

ErrorEmptyHandler.propTypes = {
  ErrorComponent: PropTypes.func,
  EmptyComponent: PropTypes.func,
  Component: PropTypes.func,
  errorMessage: PropTypes.string,
  emptyMessage: PropTypes.string,
  emptyPropertyKey: PropTypes.string,
  componentProps: PropTypes.object,
};

export default ErrorEmptyHandler;
