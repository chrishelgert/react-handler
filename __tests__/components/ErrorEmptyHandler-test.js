import React from 'react';
import renderer from 'react-test-renderer';
import ErrorEmptyHandler from '../../src/components/ErrorEmptyHandler';

const Component = () => <div>Test</div>;

describe('ErrorEmptyHandler', () => {
  it('should display the error-message', () => {
    const tree = renderer.create(
      <ErrorEmptyHandler
        Component={Component}
        errorMessage="failed to load articles"
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should display the empty-message', () => {
    const tree = renderer.create(
      <ErrorEmptyHandler
        Component={Component}
        componentProps={{}}
        emptyPropertyKey="articles"
        emptyMessage="0 Articles found"
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should display the Component', () => {
    const tree = renderer.create(
      <ErrorEmptyHandler Component={Component} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
