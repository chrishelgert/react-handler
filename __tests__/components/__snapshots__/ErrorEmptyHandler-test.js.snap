exports[`ErrorEmptyHandler should display the Component 1`] = `
<div>
  Test
</div>
`;

exports[`ErrorEmptyHandler should display the empty-message 1`] = `
<div
  className="handler--empty">
  0 Articles found
</div>
`;

exports[`ErrorEmptyHandler should display the error-message 1`] = `
<div
  className="handler--error">
  failed to load articles
</div>
`;
